sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/VectorizedHashMapGenerator.scala

```diff
@@ -75,9 +77,14 @@ class VectorizedHashMapGenerator(
           }
         }.mkString("\n").concat(";")
 
+    val columnVector = if (SparkEnv.get.memoryManager.tungstenMemoryMode == MemoryMode.ON_HEAP) {
+      "OnHeapColumnVector"
+    } else {
+      "OffHeapColumnVector"
+    }
     s"""
-       |  private org.apache.spark.sql.execution.vectorized.OnHeapColumnVector[] batchVectors;
-       |  private org.apache.spark.sql.execution.vectorized.OnHeapColumnVector[] bufferVectors;
+       |  private org.apache.spark.sql.execution.vectorized.WritableColumnVector[] batchVectors;
+       |  private org.apache.spark.sql.execution.vectorized.WritableColumnVector[] bufferVectors;
```

Comment:
We can use `$columnVector` instead of `WritableColumnVector`?

Reply:
Good catch, thanks